src/components/date-picker/date-picker.tsx

```diff
@@ -195,26 +135,63 @@ export class DatePicker {
                     helperText={this.helperText}
                     required={this.required}
                     value={this.formattedValue}
-                    onChange={this.handleChange}
+                    onFocus={this.showPortalElement}
+                    ref={(el) => (this.textField = el)}
                 />
+                <limel-portal
+                    containerId={this.portalId}
+                    visible={this.showPortal}
+                >
+                    <limel-flatpickr-adapter
+                        type={this.type}
+                        value={this.value}
+                        ref={(el) => (this.datePickerCalendar = el)}
+                        onChange={this.handleCalendarChange}
+                    />
+                </limel-portal>
             </div>
         );
     }
 
-    @Listen('resize', { target: 'window' })
-    public resizeEvent() {
-        this.picker.init(this.input, this.container, this.value);
-    }
-
     @Watch('value')
     protected onValueChange(newValue, oldValue) {
-        if (newValue !== oldValue) {
-            this.formattedValue = this.picker.formatDate(newValue);
+        if (newValue !== oldValue && newValue !== this.formattedValue) {
+            this.formattedValue = this.dateFormatter.formatDateByType(
+                newValue,
+                this.type
+            );
         }
     }
 
-    private handleChange(event) {
+    private showPortalElement(event) {
+        this.showPortal = true;
+        const inputElement = this.textField.shadowRoot.querySelector('input');
+        setTimeout(() => {
+            this.datePickerCalendar.inputElement = inputElement;
+        });
         event.stopPropagation();
-        this.formattedValue = event.detail;
+
+        document.addEventListener('mousedown', this.eventListener, { passive: true });
```

Comment:
This means that if the `mousedown` event is caught somewhere and its propagation is stopped, the portal won't be closed. I'm guessing the risk of that happening is pretty small, since we're usually using with the `click` event instead. But thought I'd mention it.

Maybe the portal could also use a scrim? One problem with a scrim is that if the menu is "dropping out of" an input field, the field will normally be underneath the scrim, and thus not possible to click on without closing the dropdown. But perhaps we can move the whole input field into the scrim in those cases? (It's currently not a problem for datepicker at the moment, since it's not possible to directly input text into the input anyway.)